### ISSUE TITLE
api function definition in multiple files

### ISSUE THREAD

**Issue by flashpixx** [rstudio/plumber#266]
Hello, I try to build a complex API, so I would like to split all functions, which defines a service in a single file. But on the server call I need to

```
server <- plumber::plumb( "myservies.r"  )
```

In my case a solution like

```
server <- plumber::plumb( "route/*.r"  )
```

which includes all r-Files with their function, is helpful to build complex APIs, I try to use the ```dir``` option, but how can I set Plumber with multiple source files of my services?

**Comment by layik** [comment]
Good question @flashpixx how does one do this? I just tried a couple of ways I thought may work but didnt. [Looking](https://github.com/trestletech/plumber/blob/master/R/plumber.R#L16) at the code, it seems it only works with an `entrypoint.R` or `plumber.R` file as stated in the docs.

Would @trestletech care to clarify this, I am happy to send a pull request to the docs clarifying this.

PS: I am new to R and plumber but I guess this is more R than a plumber issue.

**Comment by flashpixx** [comment]
In my use-case I have got a large set of functions, so on development perspective it is easier to write single files with the functions and group different files within directories. Based on this use-case it could be nice if I can pass a path to my parent directory and Plumber should read all files recursive e.g.

```
    route/
              functionset1/
                                   func1.r
                                   func2.r
              functionset2/
                                    anotherfunction.r
```

than I call ```server <- plumber::plumb( "./route"  )``` and get the REST URL like

* http://localhost:8888/functionset1/func1
* http://localhost:8888/functionset1/func2
* http://localhost:8888/functionset2/anotherfunction

This feature would be greate

**Comment by schloerke** [comment]
What about making a server programmatically by mounting each file by the file name? 

(untested code)
```r
make_server <- function(files) {
  pr <- plumber$new()

  for (file in files) {
    prFile <- plumb(file)
    path <- gsub("\\..*$", "", basename(file))
    path <- paste0("/", path)
    pr$mount(path, prFile)
  }

  pr
}

server <- make_server(dir("route", pattern = ".r", full.names = TRUE))
```

**Comment by ilarischeinin** [comment]
This has also been discussed in #36 and #56.

@schloerke's approach works (when `pr` is replaced with `pr$run()`) in terms of the API responding, but not for Swagger. You just get a message "No operations defined in spec!".

**Comment by schloerke** [comment]
swagger for mounted routes is coming in PR #274

**Comment by psramkumar** [comment]
you could use the Single file and include all the other files as below
```
if(!exists("sv_initiate_1", mode="function")) source(paste0(r_script_path, "service_definition1.r"))
if(!exists("sv_initiate_2", mode="function")) source(paste0(r_script_path, "service_definition2.r"))
```

**Comment by schloerke** [comment]
swagger docs have been merged to master (PR #280)

**Comment by schloerke** [comment]
In combination when using the `#' @plumber` to `pr_mount(pr, sub_pr)`, I think this request can be resolved.  

Ex:
```
# plumber.R

#' @get /echo
function() {
  Sys.time()
}

#' @plumber
function(pr) {
  pr %>%
    pr_mount("/my_route1", plumb("./file1/myfile.R")) %>%
    pr_mount("/my_route2", plumb("./file1/myfile2.R")) %>%
    pr_mount("/my_route3", plumb("./file2/myfile3.R")) %>%
    pr_mount("/my_route4", plumb("./file2/myfile4.R"))
}
```

I believe we'll quickly run into situations to where this doesn't scale. I'd like to take the stance of a single root file to trust and this file should determine how it's routes or mounted routers are added.
